cot/src/form/field_value.rs

```diff
@@ -0,0 +1,259 @@
+                .inner
+                .bytes()
+                .await
+                .map_err(FormFieldValueError::from_multer),
+        }
+    }
+
+    /// Converts the field value into text.
+    ///
+    /// For text fields, this returns the text directly. For multipart fields,
+    /// this reads the field's content as text.
+    ///
+    /// # Errors
+    ///
+    /// This method can return an error if the field is a multipart field and
+    /// the content cannot be read, for example, because of an I/O error.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::form::FormFieldValue;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> cot::Result<()> {
+    /// let text_value = FormFieldValue::new_text("hello");
+    /// assert_eq!(text_value.into_text().await?, "hello");
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn into_text(self) -> Result<String, FormFieldValueError> {
+        match self.inner {
+            FormFieldValueImpl::Text(text) => Ok(text),
+            FormFieldValueImpl::Multipart(multipart) => multipart
+                .inner
+                .text()
+                .await
+                .map_err(FormFieldValueError::from_multer),
+        }
+    }
+
+    /// Returns whether this is a multipart field.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::form::FormFieldValue;
+    ///
+    /// let text_value = FormFieldValue::new_text("hello");
+    /// assert!(!text_value.is_multipart());
+    /// ```
+    #[must_use]
+    pub fn is_multipart(&self) -> bool {
+        matches!(self.inner, FormFieldValueImpl::Multipart(_))
+    }
+}
+
+#[derive(Debug)]
+enum FormFieldValueImpl<'a> {
+    Text(String),
+    Multipart(Box<MultipartField<'a>>),
+}
+
+#[derive(Debug)]
+struct MultipartField<'a> {
+    inner: multer::Field<'a>,
+}
+
+/// An error that can occur when processing a form field value.
+///
+/// This type represents errors that can occur when processing form field
+/// values, such as errors from the multipart parser or validation errors.
+#[derive(Debug, PartialEq, Eq, Error)]
+#[error(transparent)]
+pub struct FormFieldValueError {
+    inner: FormFieldValueErrorImpl,
+}
+
+#[derive(Debug, PartialEq, Eq, Error)]
+enum FormFieldValueErrorImpl {
+    #[error(transparent)]
+    Multer(multer::Error),
+    #[error("Multipart field does not have a name")]
+    NoName,
+    #[error("File field requires the form to be sent as multipart/form-data")]
+    MultipartRequired,
+}
+
+impl FormFieldValueError {
+    pub(crate) fn from_multer(multer: multer::Error) -> Self {
+        Self {
+            inner: FormFieldValueErrorImpl::Multer(multer),
+        }
+    }
+
+    pub(crate) fn no_name() -> Self {
+        Self {
+            inner: FormFieldValueErrorImpl::NoName,
+        }
+    }
+
+    pub(crate) fn multipart_required() -> Self {
+        Self {
+            inner: FormFieldValueErrorImpl::MultipartRequired,
+        }
+    }
+}
```

Comment:
Why is the indirection needed here? Couldn't you merge those into a single enum?